src/Struct/FindologicProduct.php

```diff
@@ -356,11 +356,11 @@ public function getKeywords(): array
 
     protected function setKeywords(): void
     {
-        $tags = $this->product->getTags();
-        if ($tags !== null && $tags->count() > 0) {
-            foreach ($tags as $tag) {
-                if (!Utils::isEmpty($tag->getName())) {
-                    $this->keywords[] = new Keyword($tag->getName());
+        $keywords = $this->product->getSearchKeywords();
+        if ($keywords !== null && $keywords->count() > 0) {
+            foreach ($keywords as $keyword) {
+                if (!Utils::isEmpty($keyword)) {
```

Comment:
Here the `$keyword` is an object and will never be empty as it is being used inside the loop. We need to ensure the actual keyword text inside the object is not empty, so it needs to check for the keyword value.